src/main/java/org/springframework/data/couchbase/core/ReactiveFindByQueryOperationSupport.java

```diff
@@ -147,7 +148,13 @@ public ReactiveFindByQueryOperationSupport(final ReactiveCouchbaseTemplate templ
 					String id = "";
 					long cas = 0;
 					if (distinctFields == null) {
+						if( row.getString(TemplateUtils.SELECT_ID) == null){
+							throw new CouchbaseException("query did not project "+TemplateUtils.SELECT_ID+" "+statement);
```

Comment:
I think both exceptions need more explanation in the message for the user what to do and how to fix it

Reply:
fixed:
com.couchbase.client.core.error.CouchbaseException: query did not project __id. Either use #{#n1ql.selectEntity} or project __id and __cas : SELECT __cas, * from `b406ab45-ef95-441f-a318-76b8e8af2d76` where iata = $1